File: python/get_user_id.py
#-- title: Get User ID
#-- description: Code to retrieve user_id from username
#-- tags: python, medium_api, medium_api_py


# Import libraries
import os
from medium_api import Medium

#%%
# Get RAPIDAPI_KEY from the environment
api_key = os.getenv('RAPIDAPI_KEY')

#%%
# Create a `Medium` Object
medium = Medium(api_key)

#%%
# Get the `User` Object using "username" and print ID
user = medium.user(username="nishu-jain")
print(user.user_id)

File: javascript/getUserId.js
// getUserId.js - Fetches the user ID for a given username (e.g., 'nishu-jain') on Medium

import Medium from 'medium-api-js';
import dotenv from 'dotenv';

dotenv.config();

const medium = new Medium(process.env.RAPIDAPI_KEY);

// Fetch the user ID for the specified username
medium.getUserId('nishu-jain')
    .then(data => console.log('User:', data));
